--- mlme.orig.c	2026-04-28 19:17:22.999992174 -0600
+++ mlme.c	2026-05-02 19:58:16.946705060 -0600
@@ -438,6 +438,18 @@
 	ieee80211_apply_htcap_overrides(sdata, &sta_ht_cap);
 
 	/*
+	 * Similar to the issue below with VHT, some APs, mainly Comcast XB7+,
+	 * are advertizing an all-F value, meaning the AP is requiring MCS
+	 * 0-76 in order to connect with HT. Connection to the affected 2.4
+	 * and 5ghz bands will fall back to legacy 802.11a/g,even if the
+	 * hardware and regulations support VHT or HE (and presumably EHT)
+	 * HT Basic MCS set cannot be used, so check only in strict mode
+	 * as is done in the VHT section.
+	 */
+	if (!ieee80211_hw_check(&sdata->local->hw, STRICT))
+		return true;
+
+	/*
 	 * P802.11REVme/D7.0 - 6.5.4.2.4
 	 * ...
 	 * If the MLME of an HT STA receives an MLME-JOIN.request primitive
